front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -1,20 +1,34 @@
 import { useEffect, useState } from 'react';
 
+import { useTranslation } from 'react-i18next';
+
 import { STDCM_TRAIN_ID } from 'applications/stdcm/consts';
 import {
   osrdEditoastApi,
   type PathProperties,
   type PostV2InfraByInfraIdPathPropertiesApiArg,
 } from 'common/api/osrdEditoastApi';
+import { addElementAtIndex } from 'utils/array';
 
 const useGetProjectedTrainOperationalPoints = (
   trainIdUsedForProjection?: number,
   infraId?: number
 ) => {
+  const { t } = useTranslation('simulation');
   const [operationalPoints, setOperationalPoints] = useState<
     NonNullable<PathProperties['operational_points']>
   >([]);
 
+  const { data: trainScheduleUsedForProjection } =
+    osrdEditoastApi.endpoints.getV2TrainScheduleById.useQuery(
```

Comment:
Instead of making a new query, should we take the train schedule object as input parameter of this hook? I think that would be simpler and less reliant on the rtk-query cache.

Reply:
Yeah I thought it was a bit overkill too... The only other place where we have the data is when we fetch the batch trainSchedules in `ScenarioV2` component. I could make a `useMemo` watching for `trainIdUsedForProjection` and keeping track of the related `trainSchedules` and pass it to `SimulationResults` and then this custom hook.
What do you think ?

Reply:
Hm, don't we have the train schedule in `selectedTrainSchedule` in `SimulationResultsV2`? Or is this somehow sometimes not the same train as `trainIdUsedForProjection`? Or is there another reason to not use that one?

Reply:
Yep it's not that easy :) The selectedTrainSchedule (green background) is not always the same than the one used for projection (blue square around the train on the left of the card)

Reply:
Right that makes sense indeed…! If it's not too much of a hassle I'd indeed prefer to grab the projected train schedule from `trainSchedules` in `ScenarioV2` as you suggested since that'd save an HTTP request.

Reply:
Done